### PR TITLE
Only log IOExceptions to Crashlytics if there is an internet connection

### LINKED ISSUE #466: Check internet connectivity before trying to update a bank.
To improve the error handling and the user experience we should check the internet connectivity before we are trying to connect to a bank.

See http://stackoverflow.com/a/27312494/738903 for a nice implementation.

### PR DESCRIPTION
**Pull request by goober**
Fixes #466

### CODE REVIEW

**Comment by goober** [comment]
@fredrike PR updated. 

All auto format parts has been removed.
`NetworkUtils#ping` has been reimplemented using `DatagramSocket` as you suggested.

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/FirstCard.java [hunk 1]
```diff
@@ -128,45 +121,34 @@ public void update() throws BankException, LoginException, BankChoiceException {
 			if (accounts.isEmpty()) {
 				throw new BankException(res.getText(R.string.no_accounts_found).toString());
 			}			
-		} catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		} catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-        finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/FirstCard.java]: We don't need a `finally` block here. If Exceptions are thrown are not `updateComplete()` either.

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Handelsbanken.java [hunk 2]
```diff
@@ -144,43 +136,30 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-        finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Handelsbanken.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Marginalen.java [hunk 3]
```diff
@@ -156,43 +149,31 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Marginalen.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Meniga.java [hunk 4]
```diff
@@ -134,50 +117,40 @@ public void update() throws BankException, LoginException, BankChoiceException {
                 throw new BankException(res.getText(R.string.no_accounts_found).toString());
             }
         }
-        catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-        }
         finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Meniga.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/OKQ8.java [hunk 5]
```diff
@@ -222,12 +215,7 @@ public void update() throws BankException, LoginException, BankChoiceException {
 				transactions.add(new Transaction(matcher.group(1).trim(), Html.fromHtml(matcher.group(2)).toString().trim(), Helpers.parseBalance(matcher.group(3)).negate()));
 			}
 			accounts.get(0).setTransactions(transactions);
-		} catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-		} catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-		}		
-        finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/OKQ8.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Ostgotatrafiken.java [hunk 6]
```diff
@@ -164,14 +156,7 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Ostgotatrafiken.java]: Remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/PayPal.java [hunk 7]
```diff
@@ -176,16 +170,8 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
             }
-        }
-        catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        finally {
+        } finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/PayPal.java]: remove `finally { `

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/SveaDirekt.java [hunk 8]
```diff
@@ -140,17 +133,9 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
             this.setAccounts(accounts);
-        }
-        catch (ClientProtocolException e) {
-          throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        finally {
+        } finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/SveaDirekt.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/TestBank.java [hunk 9]
```diff
@@ -111,45 +112,32 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/TestBank.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/AkeliusSpar.java [hunk 10]
```diff
@@ -173,43 +166,33 @@ public void update() throws BankException, LoginException, BankChoiceException {
 						if (accounts.isEmpty()) {
 				throw new BankException(res.getText(R.string.no_accounts_found).toString());
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
 		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/AkeliusSpar.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/AkeliusInvest.java [hunk 11]
```diff
@@ -173,43 +165,32 @@ public void update() throws BankException, LoginException, BankChoiceException {
 						if (accounts.isEmpty()) {
 				throw new BankException(res.getText(R.string.no_accounts_found).toString());
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
 		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/AkeliusInvest.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/CSN.java [hunk 12]
```diff
@@ -170,80 +156,69 @@ public void update() throws BankException, LoginException, BankChoiceException {
 				throw new BankException(res.getText(R.string.no_accounts_found).toString());
 			}
 		}
-        catch (ClientProtocolException e) {
-           throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-           throw new BankException(e.getMessage(), e);
-        }		
         finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/CSN.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/DanskeBank.java [hunk 13]
```diff
@@ -185,48 +179,35 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/DanskeBank.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Vasttrafik.java [hunk 14]
```diff
@@ -151,14 +143,7 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
             }
-        }
-        catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        finally {
+        } finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Vasttrafik.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Volvofinans.java [hunk 15]
```diff
@@ -148,20 +140,14 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-		catch (ClientProtocolException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		catch (IOException e) {
-			throw new BankException(e.getMessage(), e);
-		}
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Volvofinans.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/nordea/Nordea.java [hunk 16]
```diff
@@ -313,14 +304,7 @@ public void update() throws BankException, LoginException, BankChoiceException {
...
 			}
-		}
-        catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-        catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
-        }
-		finally {
+		} finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/nordea/Nordea.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/sebkort/SEBKortBase.java [hunk 17]
```diff
@@ -179,10 +174,6 @@ public void update() throws BankException, LoginException, BankChoiceException {
             if (accounts.isEmpty()) {
                 throw new BankException(res.getText(R.string.no_accounts_found).toString());
             }
-        } catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        } catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
         } finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/sebkort/SEBKortBase.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/swedbank/AbstractSwedbank.java [hunk 18]
```diff
@@ -163,17 +157,14 @@ public void update() throws BankException, LoginException, BankChoiceException {
             if (this.accounts.isEmpty()) {
                 throw new BankException(res.getText(R.string.no_accounts_found).toString());
             }
-        } catch (ClientProtocolException e) {
-            throw new BankException(e.getMessage(), e);
-        } catch (IOException e) {
-            throw new BankException(e.getMessage(), e);
         } finally {
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/swedbank/AbstractSwedbank.java]: remove `finally {`

File: bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/icabanken/ICABanken.java [hunk 19]
```diff
@@ -130,11 +124,6 @@ public void update() throws BankException, LoginException,
 
     }
 
-    @Override
```
Comment (fredrike) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/icabanken/ICABanken.java]: Don't we need this `@Override`?

Comment (goober) [inline bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/icabanken/ICABanken.java]: This is just an empty method and the super method in Bank is not abstract so this code is redundant.

**Comment by goober** [comment]
@fredrike PR is updated with finally clauses removed

**Comment by fredrike** [comment]
Thank you!

**Pull request merged**